code/common/subscriptions.q

```diff
@@ -5,126 +5,170 @@
-		subtabs:subtabs except s`table];
-
-	/-if all the tables have been subscribed to on specified instruments
-	if[0=count subtabs; :()];
-	/-if the requested tables aren't available, ignore them and log a message
-	if[count e:subtabs except utabs;
-		.lg.o[`subscribe;"tables ",("," sv string e)," are not available to be subscribed to, they will be ignored"];
-		subtabs:subtabs inter utabs;];
-	/-subscribe and get the details for the tables
-	/-if replaylog is false,dont try to get the log file details - they may not exist
-	/-set the function to send to the server based on this
-	.lg.o[`subscribe;"getting details from the server"];
-	df:{(.u.sub\:[x;y];(.u`i`L);(.u `icounts);(.u `d))};
-	details:@[proc`w;(df;subtabs;instrs);{.lg.e[`subscribe;"subscribe failed : ",x];()}];
-	/-to be returned at end of function (null if there is no log)
-	logdate: 0Nd;
-	if[count details;
-		if[setschema;
-			.lg.o[`subscribe;"setting the schema definition"];
-			/-the first element of details is a list of pairs (tablename; schema)
-			/-this is the same as (tablename set schema)each table subscribed to
-			(@[`.;;:;].)each details[0] where not nulldetail:0=count each details 0;];
-		if[replaylog&not null details[1;1];
-			.lg.o[`subscribe;"replaying the log file"];
-			/-store the orig version of upd
-			origupd:@[value;`..upd;{{[x;y]}}];
-			/-only use tables user has access to
-			if[count where nulldetail;
-				tabs:(details[0] where not nulldetail)[;0];
-				subtabs:tabs inter subtabs];
-			/-set the replayupd function to be upd globally
-			if[not (tabs;instrs)~(`;`);
-				.lg.o[`subscribe;"using the .sub.replayupd function as not replaying all tables or instruments"];
-				@[`.;`upd;:;.sub.replayupd[origupd;subtabs;instrs]]];
-			/-the second element of details is a pair (log count;logfile)
-			@[{-11!x;};details 1;{.lg.e[`subscribe;"could not replay the log file: ", x]}];
-			/-reset the upd function back to original upd
-			@[`.;`upd;:;origupd]];
-		if[replaylog&null details[1;1];
-			.lg.e[`subscribe;"replaylog set to true but TP not using log file"]];
-		/-insert the details into the SUBSCRIPTIONS table
-		.lg.o[`subscribe;"subscription successful"];
-		updatesubscriptions[proc;;instrs]each subtabs];
-		/-return the names of the tables that have been subscribed for and
-		/-the date from the name of thr tickerplant log file (assuming the tp log has a name like `: sym2014.01.01
-		/-plus .u.i and .u.icounts if existing on TP - details[1;0] is .u.i, details[2] is .u.icounts (or null)
-		(`subtables`tplogdate!(details[0;;0];(first "D" $ -10 sublist string last details 1)^logdate)),{(where 101 = type each x)_x}(`i`icounts`d)!(details[1;0];details[2];details[3])
-	}
-
-/-wrapper function around upd which is used to only replay syms and tables from the log file that
-/-the subscriber has requested
+ // if proc dictionary is empty then exit - no connection
+ if[0=count proc;.lg.o[`subscribe;"no connections made"]; :()];
```

Comment:
Is the only change here going from tabs -> spaces? If so that should be undone as it masks "real" changes to this file.